### DISCUSSION TITLE
Restore nominatim server problem

### DISCUSSION THREAD

**Discussion by francklap** [osm-search/Nominatim#2376] [General]
Hello,

I recently installed a new Nominatim server with version 3.7.2, with 2 ZFS dataset, 1 for Postgresql and 1 for Nominatim installation. After that I installed another server, and to prevent waiting several days for the import, I restore Postgresql and Nominatim datasets from the 1st server. Then I have an error for replication saying that it can't find /data/nominatim/module/nominatim.so because it is in /data/nominatim/nominatim-planet/module/nominatim.so.
Is it the import which install this module and define its PATH? Because except import commands, I followed all the documentation. And if I create a link /data/nominatim/module to data/nominatim/nominatim-planet/module it seems to work perfectly.

Thanks for your help.

**Comment by lonvia** [comment]
You can define a custom location of the module in your `.env` file using `NOMINATIM_DATABASE_MODULE_PATH`. There is some discussion on that in the [section about running Nominatim on an external server](https://nominatim.org/release-docs/latest/admin/Advanced-Installations/#importing-nominatim-to-an-external-postgresql-database).

When you dump and restore the Nominatim database, I strongly recommend to rerun `nominatim refresh --functions` after restoring the data. This ensures that the external paths and stored procedures are updated accordingly. You can also run `nominatim admin --check-database` when everything is done to confirm that all has gone well with the restore.
